### ISSUE TITLE
Fix CI job `Generate coverage data / Generate coverage and push to Coveralls.io`

### ISSUE THREAD

**Issue by PokhodenkoSA** [IntelPython/dpctl#611]
I have made job "Generate coverage data / Generate coverage and push to Coveralls.io" as not required because it fails on master too.

_Originally posted by @PokhodenkoSA in https://github.com/IntelPython/dpctl/issues/599#issuecomment-931142866_

**Comment by PokhodenkoSA** [comment]
@diptorupd
Could you please check why it fails?

**Comment by oleksandr-pavlyk** [comment]
@diptorupd was investigating this, I looked it up too.

From the log, it appears somehow `python setup.py develop --coverage` silently crashes after coverage collection completes and never gets to build Python extension. Then it launches pytest, which fails at collection stage.

This started with api 2021.4 appearance on APT servers.

**Comment by diptorupd** [comment]
@oleksandr-pavlyk most probably the issue is related to the bug/feature in 2021.4's dpcpp_cpp_rt's `prefetch` call. Once your workaround is merged, I expect tests and coverage to start working again.

**Comment by diptorupd** [comment]
The issue is some change in how `llvm-profdata` is generating the profile.

Bare bones reproducer:

```
// hello_world.cpp
#include <iostream>

void print_msg(const char* msg)
{
    std::cout << msg << '\n';
}

int main ()
{
    print_msg("hello World");
}
```

> dpcpp -fprofile-instr-generate -fcoverage-mapping  hello_world.cpp
> llvm-profdata merge -sparse default.profraw -o hello.profdata
> llvm-cov show ./a.out --instr-profile=hello.profdata 

I can reproduce the failure by doing just this much:
```
error: /tmp/hello_world-48566a.cpp: No such file or directory
warning: The file '/tmp/hello_world-48566a.cpp' isn't covered.
```

**Comment by diptorupd** [comment]
Or it might be that dpcpp's clang some how is embedding the mangled filename information into the profile.

**Comment by oleksandr-pavlyk** [comment]
Filed as XDEPS-2775.

By the way, I tried reproducing it with OS compiler, but got:

```
(build_sycl) [08:51:27 ansatnuc04 llvm]$ clang++ -fsycl -fprofile-instr-generate -fcoverage-mapping hello.cpp
/usr/bin/ld: cannot find /localdisk/work/opavlyk/repos/llvm/build/lib/clang/14.0.0/lib/linux/libclang_rt.profile-x86_64.a: No such file or directory
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
```

I would love to know how to build the open source compiler with profile clang_rt.

**Comment by diptorupd** [comment]
Fixed by #614. We can revisit once 2022.1 (gold) is out or else keep using 2021.3 to generate coverage.
